File: interface/data/assignments/Assignment.ts
// Copyright © Spatial Corporation. All rights reserved.

import { Resource } from "..";

/**
 * A user role assignment.
 */
export type Assignment = Resource<{
  /**
   * The user the role is assigned to.
   */
  user: string;

  /**
   * The role that was assigned.
   */
  role: string;
}>;

File: cloud/Core/Identity/Authorization/Assignment.cs
// Copyright © Spatial Corporation. All rights reserved.

using Spatial.Persistence;

namespace Spatial.Identity.Authorization;

/// <summary>
/// ...
/// </summary>
[Collection("assignments")]
public class Assignment : Resource
{
    /// <summary>
    /// The user the <see cref="Role"/> is assigned to.
    /// </summary>
    public string User { get; set; }

    /// <summary>
    /// A <see cref="Role"/> identifier.
    /// </summary>
    public string Role { get; set; }
}